modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/FootprintLoaderFinder.java

```diff
@@ -0,0 +1,125 @@
+ *
+ * <p>
+ * <code>META-INF/services/org.geotools.coverage.grid.io.footprint.FootprintLoaderSpi</code>
+ * </p>
+ * *
+ * <p>
+ * Example:<br/>
+ * <code>org.geotools.coverage.grid.io.footprint.WKBLoaderSPI</code>
+ * </p>
+ * 
+ * @author Daniele Romagnoli, GeoSolutions
+ *
+ */
+public final class FootprintLoaderFinder {
+    /**
+     * The service registry for this manager. Will be initialized only when first needed.
+     */
+    private static FactoryRegistry registry;
+
+    /**
+     * Do not allows any instantiation of this class.
+     */
+    private FootprintLoaderFinder() {
+        // singleton
+    }
+
+    /**
+     * Finds all available implementations of {@link FootprintLoaderSpi} which have registered 
+     * using the services mechanism.
+     *
+     * @return An unmodifiable {@link Set} of all discovered modules which have registered factories
+     */
+    public static synchronized Set<FootprintLoaderSpi> getAvailableLoaders() {
+        // get all FootprintLoaderSpi implementations
+        scanForPlugins();
+        final Iterator<FootprintLoaderSpi> it = getServiceRegistry().getServiceProviders(
+                FootprintLoaderSpi.class, true);
+        final Set<FootprintLoaderSpi> loaders = new HashSet<FootprintLoaderSpi>();
+        while (it.hasNext()) {
+            loaders.add(it.next());
+        }
+        return Collections.unmodifiableSet(loaders);
+    }
+
+    /**
+     * Returns the service registry. The registry will be created the first time this method is invoked.
+     */
+    private static FactoryRegistry getServiceRegistry() {
+        assert Thread.holdsLock(FootprintLoaderFinder.class);
+        if (registry == null) {
+            registry = new FactoryCreator(Arrays.asList(new Class<?>[] { FootprintLoaderSpi.class }));
+        }
+        return registry;
+    }
+
+    /**
+     * Scans for factory plug-ins on the application class path. This method is needed 
+     * because the application class path can theoretically change, or additional plug-ins 
+     * may become available. Rather than re-scanning the classpath on every invocation of 
+     * the API, the class path is scanned automatically only on the first invocation. 
+     * Clients can call this method to prompt a re-scan. Thus this method need only be invoked by
+     * sophisticated applications which dynamically make new plug-ins available at runtime.
+     */
+    public static synchronized void scanForPlugins() {
+        getServiceRegistry().scanForPlugins();
+    }
+
+    /**
+     * Returns an array with all available {@link FootprintLoaderSpi} implementations.
+     *
+     * <p>
+     * It can be used to get basic information about all the available {@link FootprintLoader} plugins. 
+     * Note that this method finds all the implemented plugins.
+     *
+     * @return an array with all available {@link FootprintLoaderSpi} implementations.
+     */
+    public static FootprintLoader[] getFootprintLoaderArray() {
```

Comment:
Weird to have a method explicitly to return an array... I would have left a list? Also, I don't see anything using it.